src/http/routing.rs

```diff
@@ -496,4 +499,69 @@ routes! ('a, {
     StageInstance { channel_id: ChannelId },
     api!("/stage-instances/{}", channel_id),
     Some(RatelimitingKind::Path);
+
+    OAuth2Token {
+        state: Option<&'a str>,
+        grant_type: GrantType,
+        code: Option<&'a str>,
+        redirect_uri: Option<&'a Url>,
+        refresh_token: Option<&'a str>,
+        scope: Vec<Scope>
+    },
+    api!(
+        "/oauth2/token?{}grant_type={}{}{}{}{}",
+        if let Some(state) = state {
+            format!("state={state}&")
+        } else {
+            String::new()
+        },
+        grant_type,
+        if let Some(code) = code {
+            format!("&code={code}")
+        } else {
+            String::new()
+        },
+        if let Some(redirect_uri) = redirect_uri {
+            format!("&redirect_uri={redirect_uri}")
+        } else {
+            String::new()
+        },
+        if let Some(refresh_token) = refresh_token {
+            format!("&refresh_token={refresh_token}")
+        } else {
+            String::new()
+        },
+        if scope.is_empty() {
+            String::new()
+        } else {
+            format!(
+                "&scope={}",
+                scope
+                    .into_iter()
+                    .map(|scope| scope.to_string())
+                    .collect::<Vec<String>>()
+                    .join(" "),
+            )
+        },
+    ),
+    None;
+
+    OAuth2TokenRevocation {
+        token: &'a str,
+        token_type_hint: Option<TokenTypeHint>
+    },
+    api!(
+        "/oauth2/token/revoke?token={}{}",
+        token,
+        if let Some(token_type_hint) = token_type_hint {
+            format!("&token_type_hint={token_type_hint}")
+        } else {
+            String::new()
+        },
+    ),
+    None;
+
+    OAuth2AuthorizationCurrent,
+    api!("/oauth2/@me"),
+    None;
```

Comment:
Query params for endpoints shouldn't be defined in this way. Instead, they should be passed using the `Request::params` field, see [example](https://github.com/serenity-rs/serenity/blob/current/src/http/client.rs#L2749-L2782).

Reply:
Oh whoops, wish I knew that before lol, that makes this PR *way* simpler